python/ray/data/_internal/block_list.py

```diff
@@ -15,19 +15,26 @@ class BlockList:
     change after execution due to block splitting.
     """
 
-    def __init__(self, blocks: List[ObjectRef[Block]], metadata: List[BlockMetadata]):
+    def __init__(
+        self,
+        blocks: List[ObjectRef[Block]],
+        metadata: List[BlockMetadata],
+        consumable: bool = False,
```

Comment:
The two alternatives I like best at this point are:
- `owned_by_consumer`--- pretty clear that this block list can be deleted by the consumer
- `multiple_owners` / `is_shared`--- also clear that it's shared and hence can't be deleted

Reply:
Thanks for suggestion, made it so!